packages/network/src/logic/Node.ts

```diff
@@ -534,6 +542,19 @@ export class Node extends EventEmitter {
         }
     }
 
+    private startConnectionCleanUpInterval(interval: number): NodeJS.Timeout {
+        return setInterval(() => {
+            const peerIds = this.nodeToNode.getAllConnectionNodeIds()
+            const unusedConnections = peerIds.filter((peer) => !this.isNodePresent(peer))
```

Comment:
is it possible this cleans up nodes that are involved in a currently executing instruction?
I wonder if this should be invoked in a debounced function that's called after each tracker instruction is completed, rather than on an interval?

Reply:
That could be possible. However, when churn testing locally with 40 nodes, 5 minutes of churn, and a 1 minute interval on the clean up function the the network operated as expected and did not gather high numbers of unused connections as described in the ticket.